assets/terraform-modules/bare-metal/flatcar-linux/kubernetes/controller.tf

```diff
@@ -10,6 +10,17 @@ module "controller" {
   apiserver              = format("%s.%s", var.cluster_name, var.k8s_domain_name)
   ca_cert                = module.bootkube.ca_cert
   kubelet_labels         = lookup(var.node_specific_labels, var.controller_names[count.index], {})
-  clc_snippets           = lookup(var.clc_snippets, var.controller_names[count.index], [])
-  set_standard_hostname  = true
+  set_standard_hostname  = false
+  clc_snippets = concat(lookup(var.clc_snippets, var.controller_names[count.index], []), [
+    <<EOF
+storage:
```

Comment:
We should make sure user can still override that.

Reply:
You mean the user would use a different name for the machine when defining it in the configuration and setting up a domain name than the hostname would be? I think it's good to check first what the behavior of CLC snippets is when the same file gets specified by the user again and if it doesn't work, document that the user has to set the hostname through a one-time service unit.

Reply:
Since this is a bug fix and does not change something related to how the user could overwrite the hostname, I think we should merge.